javascript/javascript1/week4/character_frequency/character.js

```diff
@@ -0,0 +1,20 @@
+const characters = [];
+function getCharacterFrequencies(inputString) {
+    for (var i = 0; i < inputString.length; i++) {
+        let freq = new Object();
+        let char = inputString.charAt(i);
+        let objectInArray = characters.find(item => item.character === char);
+        if (objectInArray) {
+            objectInArray.count++;
+        }
+
+        else {
+            freq.character = char;
+            freq.count = 1;
+            characters.push(freq);
+        }
+    }
+    return characters;
```

Comment:
ok this is really confusing, you are using `characters` declared outside the function, but also returning it. 
Technically you're allowed, but it's better to either: 
1) modify the _global_ `characters` array and not return
2) Create `characters` inside the function and then return. In this case you should be consuming the return value somewhere after the function.